pkg/api/composition.go

```diff
@@ -107,6 +108,18 @@ type Build struct {
 	Dependencies Dependencies `toml:"dependencies" json:"dependencies"`
 }
 
+// BuildKey returns a composite key that identifies this build, suitable for
+// deduplication.
+func (b Build) BuildKey() string {
+	var sb strings.Builder
+	sb.WriteString(fmt.Sprintf("selectors=%s;", strings.Join(b.Selectors, ",")))
```

Comment:
Maybe these can be sorted? It may not be obvious that selectors ["foo", "bar"] is different than ["bar", "foo"]

Reply:
Yeah! This and also the dependencies need to be sorted for a canonical key. Good catch!